pkg/assembler/backends/inmem/pkg.go

```diff
@@ -774,6 +774,7 @@ func (c *demoClient) exactPackageVersion(filter *model.PkgSpec) (*pkgVersionNode
 		}
 		for _, v := range nm.versions {
 			if *filter.Version != v.version ||
+				filter.Subpath == nil ||
 				noMatchInput(filter.Subpath, v.subpath) ||
```

Comment:
first this should be `noMatch` and not `noMatchInput`. Also, there is a logic issue here. `nm.versions` does contain the version that contains the subpath but that is never evaluated and it only returns before it can be evaluated.

Reply:
Seems like `exactPackageVersion` should return a list of matching `pkgVersionNode` that can be evaluated

Reply:
> Seems like `exactPackageVersion` should return a list of matching `pkgVersionNode` that can be evaluated as it currently returns on the first matching, when in reality it should return all matching and let the downstream sort out the ones that are matching.

I've introduced `findPackageVersion` func to return the list of matching `pkgVersionNode` and refactored the `exactPackageVersion` func for consistency with its usages done in other components (and managing the `exact` constraint accordingly)